rebasehelper/plugins/build_log_hooks/files.py

```diff
@@ -39,6 +39,9 @@
 
 logger: CustomLogger = cast(CustomLogger, logging.getLogger(__name__))
 
+AddedFiles = Dict[str, Dict[str, List[str]]]
+RemovedFiles = Dict[str, Union[List[str], Dict[str, List[str]]]]
```

Comment:
I'd rather see this in `types.py`, on the other hand, it's local to this build hook (which begs a question - if it's local, why does it need to be typed?).

Reply:
Exactly my thoughts, since it's local, `types.py` wouldn't make much sense but putting it next to the variable (or function) would make the code quite unreadable. 

I don't exactly know the reason why mypy requires typing in such cases but it seems to me that variables that are returned from methods have to be typed or their types have to be obvious (which isn't the case with lists and dicts).

Reply:
Ok, but the method is internal. The only "public" method is `run()`, which returns `Dict[Any, Any]` (and that doesn't seem right either).

Reply:
Apparently mypy doesn't differentiate between internal and external methods. Perhaps we could just ignore the error since it's internal. I think that the return value of `run` should be `Union[AddedFiles, RemovedFiles]` with that it would even make sense to have types in the internal methods. However mypy throws a different error:
```
rebasehelper/plugins/build_log_hooks/files.py:337: error: Incompatible types in assignment (expression has type "Dict[<nothing>, <nothing>]", variable has type "Union[Dict[str, Dict[str, List[str]]], Dict[str, Union[List[str], Dict[str, List[str]]]]]")```

I will try to find a way how to fix it

Reply:
I fixed the type of the returned dict and added typing to the `run` method. I can't find a way to make mypy ignore the internal methods, it just checks everything...